2-mandatory/2-water-bottle.js

```diff
@@ -19,18 +19,33 @@ let bottle = {
   volume: 0,
   fillUp: function () {
     // calling this function should completely fill your bottle (volume = 100);
+    this.volume = 100;
   },
   pour: function () {
     // calling this function should increase your bottle volume by 10 units;
+    if (this.volume <= 90) {
+      this.volume += 10;
+    }
   },
   drink: function () {
     // calling this function should decrease your bottle volume by 10 units;
+    if (this.volume >= 10) this.volume -= 10;
   },
   isFull: function () {
     // this function should return true if your bottle is full;
+    if (this.volume === 100) {
+      return true;
+    } else {
+      return false;
+    }
   },
   isEmpty: function () {
     // this function should return true if your bottle is empty;
+    if (this.volume === 0) {
+      return true;
+    } else {
+      return false;
+    }
```

Comment:
This all works well. For the full and empty methods you could just use "return this.volume === 100" and "return this.volume === 0"